src/controllers/movie.ts

```diff
@@ -0,0 +1,66 @@
+import { Request, Response } from 'express';
+import { Movie } from "../models";
+import { IMovie } from '../interfaces';
+import { isObjectIdOrHexString } from 'mongoose';
+
+export async function getMovies(req: Request, res: Response) {
+  return res.status(200).json(await Movie.find({}));
+}
+
+export async function getMoviesByGenre(req: Request<{ genreName: String }>, res: Response) {
+  const { genreName } = req.params;
+
+  return res.json(await Movie.find({ genre: genreName }));
+}
+
+export async function getMovieById(req: Request<{ id: String }>, res: Response) {
+  const { id } = req.params;
+
+  if (!isObjectIdOrHexString(id)) return res.status(400).json('Invalid ID');
+
+  const movie = await Movie.findById(id);
+
+  if (!movie) return res.status(404).json('Not found')
+
+  return res.status(200).json(movie);
+}
+
+export async function createMovie(req: Request<{}, {}, IMovie>, res: Response) {
+  const movie = new Movie(req.body);
+
+  const validation = movie.validateSync();
```

Comment:
Architecturally, validation can be moved to a separate middleware for better structuring and code reuse. This will improve the readability of your code and make it more modular.

Reply:
Thanks for the recommendation. I will make changes in the near future. I forgot that it was possible to carry out validation with middlewares.